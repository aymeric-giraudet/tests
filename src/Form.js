import React from "react";
import { TextInput } from "@axa-fr/react-toolkit-form-input-text";
import Button from "@axa-fr/react-toolkit-button";

const Form = ({ form, greetings, onChange, onClick }) => (
  <>
    Veuillez entrer votre nom complet :
    <TextInput
      label="PrÃ©nom"
      name="firstName"
      id="firstName"
      value={form.firstName}
      onChange={onChange}
    />
    <TextInput
      label="Nom"
      name="lastName"
      id="lastName"
      value={form.lastName}
      onChange={onChange}
    />
    <Button onClick={onClick}>Envoyer</Button>
    <p>{greetings}</p>
  </>
);

export default Form;
