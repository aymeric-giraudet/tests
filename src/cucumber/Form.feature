Feature: Filling form

Scenario: Entering your full name
    Given i am on the form
    When i enter my first name as Aymeric
    And i enter my last name as Giraudet
    And i click on the button
    Then i should be greeted as Aymeric Giraudet

Scenario: Entering your full name as John Doe
    Given i am on the form
    When i enter my first name as John
    And i enter my last name as Doe
    And i click on the button
    Then i should be greeted as John Doe