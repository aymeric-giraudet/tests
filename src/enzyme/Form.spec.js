import React from "react";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";
import Form from "../Form";

const form = {
  firstName: "Aymeric",
  lastName: "Giraudet",
};
const onChangeMock = jest.fn();
const onClickMock = jest.fn();

it("should render correctly", () => {
  const wrapper = create(
    <Form
      form={form}
      greetings="Bonjour Aymeric Giraudet !"
      onChange={onChangeMock}
      onClick={onClickMock}
    />
  );
  expect(wrapper.toJSON()).toMatchSnapshot();
});

it("should call onChange when firstName has been changed", () => {
  const wrapper = shallow(
    <Form form={form} onChange={onChangeMock} onClick={onClickMock} />
  );
  wrapper
    .find('[name="firstName"]')
    .simulate("change", { target: { value: "Aymerico" } });
  expect(onChangeMock).toHaveBeenCalledWith({
    target: {
      value: "Aymerico",
    },
  });
});

it("should call onChange when lastName has been changed", () => {
  const wrapper = shallow(
    <Form form={form} onChange={onChangeMock} onClick={onClickMock} />
  );
  wrapper
    .find('[name="lastName"]')
    .simulate("change", { target: { value: "G-Rod" } });
  expect(onChangeMock).toHaveBeenCalledWith({
    target: {
      value: "G-Rod",
    },
  });
});

it("should call onClick when button has been clicked", () => {
  const wrapper = shallow(
    <Form form={form} onChange={onChangeMock} onClick={onClickMock} />
  );
  wrapper.find("Button").simulate("click");
  expect(onChangeMock).toHaveBeenCalled();
});
