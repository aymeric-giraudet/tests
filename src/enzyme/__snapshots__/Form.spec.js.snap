// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  "Veuillez entrer votre nom complet :",
  <div
    className="row af-form__group"
  >
    <div
      className="col-md-2"
    >
      <label
        className="af-form__group-label"
        htmlFor="firstName"
      >
        Pr√©nom
      </label>
    </div>
    <div
      className="col-md-10"
    >
      <div
        className="af-form__text"
      >
        <input
          autoFocus={null}
          className="af-form__input-text"
          disabled={false}
          id="firstName"
          name="firstName"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder={null}
          readOnly={false}
          tabIndex={null}
          type="text"
          value="Aymeric"
        />
      </div>
      <small
        className="af-form__help"
      />
    </div>
  </div>,
  <div
    className="row af-form__group"
  >
    <div
      className="col-md-2"
    >
      <label
        className="af-form__group-label"
        htmlFor="lastName"
      >
        Nom
      </label>
    </div>
    <div
      className="col-md-10"
    >
      <div
        className="af-form__text"
      >
        <input
          autoFocus={null}
          className="af-form__input-text"
          disabled={false}
          id="lastName"
          name="lastName"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder={null}
          readOnly={false}
          tabIndex={null}
          type="text"
          value="Giraudet"
        />
      </div>
      <small
        className="af-form__help"
      />
    </div>
  </div>,
  <button
    className="btn af-btn"
    onClick={[Function]}
  >
    Envoyer
  </button>,
  <p>
    Bonjour Aymeric Giraudet !
  </p>,
]
`;
